tests/Integration/Subscriber/Optimization/DequeueJQueryMigrateSubscriber/dequeueJqueryMigrate.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\Subscriber\Optimization\Dequeue_JQuery_Migrate_Subscriber;
+
+use Brain\Monkey;
+use WP_Rocket\Tests\Integration\TestCase;
+
+/**
+ * @covers WP_Rocket\Subscriber\Optimization\Dequeue_JQuery_Migrate_Subscriber::dequeue_jquery_migrate
+ *
+ * @group  jQueryMigrate
+ */
+class Test_DequeueJqueryMigrate extends TestCase {
+	private $orig_deps;
+	private $wp_scripts;
+	private $count;
+
+	public function setUp() {
+		parent::setUp();
+
+		$wp_scripts       = wp_scripts();
+		$this->orig_deps  = $wp_scripts->registered['jquery']->deps;
+		$this->wp_scripts = $wp_scripts;
+		$this->count      = did_action( 'wp_default_scripts' );
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		// Restore.
+		wp_scripts()->registered['jquery']->deps = $this->orig_deps;
+	}
+
+	public function testShouldDequeueJqueryMigrate() {
+		add_filter( 'pre_get_rocket_option_dequeue_jquery_migrate', '__return_true' );
+
+		$this->wp_scripts->init();
+		$this->assertGreaterThan( $this->count, did_action( 'wp_default_scripts' ) );
+		$jquery_dependencies = $this->wp_scripts->registered['jquery']->deps;
+
+		$this->assertNotContains( 'jquery-migrate', $jquery_dependencies );
```

Comment:
```suggestion
		$this->assertFalse( 'jquery-migrate', wp_scripts()->query( 'jquery-migrate', 'enqueued' );
```

How about this instead? This method is available from this parent: https://developer.wordpress.org/reference/classes/wp_dependencies/